components/Article/Page.js

```diff
@@ -240,6 +250,74 @@ class ArticlePage extends Component {
         secondaryNavExpanded: expanded
       })
     }
+
+    this.getGalleryItems = memoize(() => {
+      const { data: { article } } = this.props
+      const shouldInclude = (el) =>
+        get(el, 'identifier', '') === 'FIGURE' && get(el, 'data.excludeFromGallery', false) !== true
+      function findFigures (node, acc = []) {
+        if (get(node, 'children.length', 0) > 0) {
+          node.children.forEach(
+            c => {
+              if (shouldInclude(c)) {
+                acc.push(c)
+              } else {
+                findFigures(c, acc)
+              }
+            }
+          )
+        }
+        return acc
+      }
+      const getImageProps = (node) => {
+        const src = get(node, 'children[0].children[0].url', '')
+        const alt = get(node, 'children[0].children[0].alt', '')
+        const caption = get(node, 'children[1].children[0].value', '')
+        const credit = get(node, 'children[1].children[1].children[0].value', '')
+        return {
+          src,
+          alt,
+          caption,
+          credit
+        }
+      }
+      return findFigures(article.content)
+        .map(getImageProps)
+        .filter(i => imageSizeInfo(i.src).width > 600)
+    })
+
+    this.renderGallery = () => {
```

Comment:
I wonder whether all new code here (including context) should be moved to a separate file to keep file length down. But I might not see any complications out of this yet.

Reply:
You are right, of course. It's much nicer to encapsulate the whole thing in a separate component!